File: 1_PythonServer/app/domain/tools/toolset.py
from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class Toolset:
    category_id: str
    name: str
    scope: str
    root_path: str
    readonly: bool
    created_at: str
    updated_at: str


@dataclass(frozen=True, slots=True)
class ToolFolder:
    project_id: str
    category_id: str
    name: str
    root_path: str
    created_at: str
    updated_at: str

File: 2_ReactWeb/src/entities/tool/model/toolset.ts
export type ToolsetScope = "local";

export type Toolset = {
  category_id: string;
  name: string;
  scope: ToolsetScope;
  root_path: string;
  readonly: boolean;
  created_at: string;
  updated_at: string;
};

export type ToolsetListResponse = {
  count: number;
  items: Toolset[];
};

export type ToolsetCreateRequest = {
  name?: string | null;
};

export type ToolsetRenameRequest = {
  name: string;
};

export type ToolFolder = {
  project_id: string;
  category_id: string;
  name: string;
  root_path: string;
  created_at: string;
  updated_at: string;
};

export type ToolFolderListResponse = {
  count: number;
  items: ToolFolder[];
};

export type ToolFolderCreateRequest = {
  name?: string | null;
};

export type ToolFolderRenameRequest = {
  name: string;
};

export type ToolFolderMoveRequest = {
  target_category_id: string;
};

export type ToolFolderRuntimeSettingsRequest = {
  enabled?: boolean;
  dynamic?: boolean;
  parallel?: boolean;
};

export type ToolFolderRuntimeSettingsResponse = {
  category_id: string;
  project_id: string;
  enabled: boolean;
  dynamic: boolean;
  parallel: boolean;
  updated_at: string;
};
